kotlin/send-http-request/Index.kt

```diff
@@ -0,0 +1,80 @@
+import com.google.gson.Gson
+import okhttp3.Headers.Companion.toHeaders
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+
+val client = OkHttpClient()
+val gson = Gson()
+
+val HTTP_METHODS =
+    listOf(
+        "GET",
+        "POST",
+        "PUT",
+        "DELETE",
+        "PATCH",
+    )
+
+data class Payload(
+    val method: String,
+    val url: String,
+    val headers: Map<String, String>,
+    val body: String?,
+)
+
+fun validatePayload(payload: Payload): Boolean {
+    if (payload.method !in HTTP_METHODS) {
+        return false
+    }
+
+    if (payload.url.isBlank()) {
+        return false
+    }
+
+    return true
+}
+
+@Throws(Exception::class)
+fun main(req: RuntimeRequest, res: RuntimeResponse): RuntimeResponse {
+    try {
+        val json: Map<String, Any> =
+            gson.fromJson<Map<String, Any>>(req.payload, MutableMap::class.java)
+                ?: return res.json(mapOf("success" to false, "message" to "Invalid json data"))
+        val headers: Map<String, String> =
+            gson.fromJson<Map<String, String>>(json["headers"].toString(), MutableMap::class.java)
+                ?: mapOf()
+        val payload =
+            Payload(
+                method = json["method"].toString(),
+                url = json["url"].toString(),
+                headers = headers,
+                body = if (json["method"].toString() != "GET") json["body"].toString() else null,
```

Comment:
I tried testing with:

```json
{"headers": {"Content-Type": "application/json"},"url": "https://jsonplaceholder.typicode.com/posts", "method": "POST", "body": "{\"title\": \"foo\"}"}
```

and

```json
{"headers": {"Content-Type": "application/json"},"url": "https://jsonplaceholder.typicode.com/posts", "method": "POST", "body": {"title": "foo"}}
```

but I'm getting the following error: 

```json
{"success":false,"message":"JSON payload parsing error: com.google.gson.stream.MalformedJsonException: Unterminated object at line 1 column 27 path $."}
```

Any idea what could be wrong?